npm/pkg/dataplane/dataplane.go

```diff
@@ -78,7 +81,26 @@ func (dp *DataPlane) BootupDataplane() error {
 
 // RunPeriodicTasks runs periodic tasks. Should only be called once.
 func (dp *DataPlane) RunPeriodicTasks() {
-	dp.policyMgr.Reconcile(dp.stopChannel)
+	go func() {
+		ticker := time.NewTicker(time.Minute * time.Duration(reconcileTimeInMinutes))
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-dp.stopChannel:
+				return
+			case <-ticker.C:
+				// locks ipset manager
+				dp.ipsetMgr.Reconcile()
+
+				// in Windows, does nothing
+				// in Linux, locks policy manager but can be interrupted
+				dp.policyMgr.Reconcile()
+
+				metrics.SendHeartbeatLog()
```

Comment:
Can we have the heart beat at the very top and in a gofunc, so it wont block the setMgr or pMgr reconcile.